hibernate6-jpa/src/test/groovy/io/micronaut/configuration/hibernate6/jpa/JpaSetupSpec.groovy

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.hibernate6.jpa
+
+import io.micrometer.core.instrument.FunctionCounter
+import io.micrometer.core.instrument.MeterRegistry
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.env.Environment
+import io.micronaut.http.exceptions.HttpException
+import io.micronaut.transaction.TransactionDefinition
+import io.micronaut.transaction.annotation.TransactionalAdvice
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+import org.hibernate.Session
+import spock.lang.AutoCleanup
+import spock.lang.Shared
+import spock.lang.Specification
+
+import jakarta.persistence.AttributeConverter
+import jakarta.persistence.Convert
+import jakarta.persistence.Entity
+import jakarta.persistence.EntityManager
+import jakarta.persistence.EntityManagerFactory
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+
+import javax.validation.ConstraintViolation
+import javax.validation.ConstraintViolationException
+import javax.validation.constraints.NotBlank
+
+/**
+ * @author graemerocher
+ * @since 1.0
+ */
+class JpaSetupSpec extends Specification {
+
+    @Shared @AutoCleanup ApplicationContext applicationContext = ApplicationContext.run(
+            'datasources.default.name':'mydb',
+            'jpa.default.properties.hibernate.hbm2ddl.auto':'create-drop',
+            'jpa.default.properties.hibernate.generate_statistics':true,
+            'micronaut.metrics.binders.hibernate.tags.some':'bar'
+    )
+
+    void "test setup entity manager with validation"() {
+        when:
+        EntityManagerFactory entityManagerFactory = applicationContext.getBean(EntityManagerFactory)
+
+        then:
+        entityManagerFactory != null
+
+        when:
+        EntityManager em = entityManagerFactory.createEntityManager()
+        def tx = em.getTransaction()
+        tx.begin()
+        em.persist(new Book(title: ""))
+        em.flush()
+        // TODO: persist/flush don't throw validation exception so do it manually
```

Comment:
Validation not triggered and tests expect it and expect ConstraintValidationException. Any idea what is changed? I tested for javax.validation.Validator and there is `MicronautHibernateValidator` but it's not being called, not sure why.

Reply:
Because it's `javax` validator, now we have `jakarta`. I think this is expected.

Reply:
Yes, `BeanValidationIntegrator` checks for existence of given class and does not find it 
```
org.hibernate.boot.registry.classloading.spi.ClassLoadingException: Unable to load class [jakarta.validation.ConstraintViolation]
```
So, what do we do with this? Are we going to get jakarta validation api through micronaut-validator future version? Can we release this without validator (in case we fix proxy factory issue)

Reply:
Disabled test with validation and added couple of TODOs so we can add validation back with jakarta api.